### DISCUSSION TITLE
DataFetchingEnvironment.getContext() deprecation

### DISCUSSION THREAD

**Discussion by bbellovic** [graphql-java-kickstart/graphql-spring-boot#747] [Q&A]
I used to use custom `GraphQLServletContext` to hold request-scoped variables (`context.getTenantId()`) like this:
```java
    public Room createRoom(final CreateRoomInput input, final DataFetchingEnvironment environment) {
        final CustomGraphQLContext context = environment.getContext();
        return roomService.create(context.getTenantId(), input);
    }
```
However `environment.getContext()` method is new deprecated and context returned from `environment.getGraphQlContext()` cannot be casted to `CustomGraphQLContext`. Is there new/another way how can I store and retrieve request-scoped variables?

**Comment by nort3x** [comment]
i also want to add emphasis on how to avoid relaying on implicit cast as it's a generic type method
consider this as well:

```kotlin
// kotlin
fun createRoom(input: CreateRoomInput, environment: DataFetchingEnvironment): Room {

    environment.getContext<CustomGraphQLContext>().apply{
            roomService.create(it.getTenantId(), input)
    }
}
``` 
```java
// java
public Room createRoom(final CreateRoomInput input, final DataFetchingEnvironment environment) {
        return roomService.create( environment.getContext<CustomGraphQLContext>().getTenantId(), input);
}
```

if we inspect `DataFetchingEnvironment` interface we can see that `getContext()` is generic (and unbounded, i really don't know what types it accepts! i can't find anything in docs as well) and `getGraphQlContext()` returns something which is really just a map

i want to know how can i get access to `HttpServletRequest`
currently i'm using this:
```kotlin
   env.getContext<GraphQLServletContext>().httpServletRequest
                // arbitrary calls to request to demonstrate my usage, i'm using it to authenticate for instance 
                .getSession(true)
                .setAttribute("SPRING_SECURITY_CONTEXT", SecurityContextHolder.getContext());

```
otherwise how can i do that?
